sdk/eventhub/Azure.Messaging.EventHubs/samples/Program.cs

```diff
@@ -0,0 +1,229 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Samples.Infrastructure;
+
+namespace Azure.Messaging.EventHubs.Samples
+{
+    /// <summary>
+    ///   The main entry point for executing the samples.
+    /// </summary>
+    ///
+    public static class Program
+    {
+        /// <summary>
+        ///   Serves as the main entry point of the application.
+        /// </summary>
+        ///
+        /// <param name="args">The set of command line arguments passed.</param>
+        ///
+        public static async Task Main(string[] args)
+        {
+            // Parse the command line arguments to ensure that there is enough information available to
+            // run samples.  If not, or if help was explicitly requeted, print the usage text and move on.
+
+            var parsedArgs = ParseArguments(args);
+
+            if ((parsedArgs.Help) || (String.IsNullOrEmpty(parsedArgs.ConnectionString)) || (String.IsNullOrEmpty(parsedArgs.EventHub)))
+            {
+                DisplayHelp();
+                return;
+            }
+
+            var samples = LocateSamples();
+
+            // Display the set of available samples and allow them to be run.
+
+            Console.WriteLine();
```

Comment:
We did the samples in App Configuration as tests https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/appconfiguration/Azure.ApplicationModel.Configuration/samples and I see here we are doing it as console app and then we run tests around them to make sure they work.
Is this the way we want to go with all samples? or why you went this route?

Reply:
My goal was:

- Avoid having magic environment variables that have to exist for samples to work; the configuration for the samples in the console approach is explicit and can be passed in whatever way a developer finds easiest.

- Allow developers to launch a sample interactively for debugging; the test approach works here as well.

- Avoid having samples need to pull in test infrastructure.

- Allow samples to be consumed by the existing test infrastructure and run as smoke tests, using the dynamic Azure resource approach used by the rest of the Live tests.


I didn't feel that the test-based approach met those goals, and I also felt that it wasn't as good of an experience for developers.